packages/react/src/components/Box/Box.tsx

```diff
@@ -1,49 +1,70 @@
 import * as React from 'react'
-import {
-  childrenExist,
-  createShorthandFactory,
-  UIComponentProps,
-  ContentComponentProps,
-  ChildrenComponentProps,
-  commonPropTypes,
-  rtlTextContainer,
-} from '../../lib'
-import createComponent, { CreateComponentReturnType } from '../../lib/createComponent'
-import { ReactProps } from '../../types'
-
-export interface BoxProps
-  extends UIComponentProps<BoxProps>,
-    ContentComponentProps,
-    ChildrenComponentProps {}
+import { createShorthandFactory, callable, getUnhandledProps } from '../../lib'
+
+import { BoxClassName, BoxProps, boxPropTypes, renderBox } from './common'
+import { FelaTheme } from 'react-fela'
+import { IRenderer as FelaRenderer } from 'fela'
+import cx from 'classnames'
+
+const withThemeContext = render => <FelaTheme>{(theme: any) => render(theme)}</FelaTheme>
+
+type BoxRenderConfig = {
+  ElementType: React.ReactType
+  props: BoxProps
+  renderer: FelaRenderer
+}
+
+const render = ({ ElementType, props, renderer }: BoxRenderConfig) => {
+  const { styles, className: predefinedClasses, content, children } = props
+
+  const unhandledProps = getUnhandledProps({ handledProps }, props)
+  const classes = {
+    root: renderer.renderRule(callable(styles), props),
+  }
+  classes.root = cx(BoxClassName, predefinedClasses, classes.root)
+
+  return renderBox({
+    ElementType,
+    classes,
+    children,
+    content,
+    unhandledProps,
+  })
+}
 
 /**
  * A Box is a basic component (no default styles)
  */
-const Box: CreateComponentReturnType<ReactProps<BoxProps>> = createComponent<BoxProps>({
-  displayName: 'Box',
-
-  className: 'ui-box',
-
-  propTypes: {
-    ...commonPropTypes.createCommon(),
-  },
-
-  render(config, props) {
-    const { ElementType, classes, unhandledProps } = config
-    const { children, content } = props
-
-    return (
-      <ElementType
-        {...rtlTextContainer.getAttributes({ forElements: [children, content] })}
-        {...unhandledProps}
-        className={classes.root}
-      >
-        {childrenExist(children) ? children : content}
-      </ElementType>
+const Box: React.FunctionComponent<BoxProps> & { create: Function } = props => {
```

Comment:
nit: ~~`React.FunctionComponent`~~ -> `React.FC`

Reply:
would rather leave it as it is, due to its readability

Reply:
Please avoid all abbreviations in the repo.  Code is for humans to read.  Let minifiers and uglifiers make it small and unreadable.